### ISSUE TITLE
too many quotes breaks SLIMERJSLAUNCHER logic in windows batch file

### ISSUE THREAD

**Issue by hallvors** [laurentj/slimerjs#348]
slimerjs.bat adds double quotes when reading the SLIMERJSLAUNCHER environment variable:

    SET SLIMERJSLAUNCHER="%SLIMERJSLAUNCHER%"

(since https://github.com/laurentj/slimerjs/commit/9e1efa886c1041146e3ebe0a3183343f40801430 )

The point of quoting it is so that you don't need to add quotes when testing for existence and actually launching the executable. And indeed, when launching the executable no quotes are used:

    %SLIMERJSLAUNCHER% -app "%SLIMERDIR%application.ini" %PROFILE% -attach-console -no-remote %__SLIMER_ARGS%

However, this un-quoted command line will turn into an error if SLIMERJSLAUNCHER has spaces and does not contain quotes as part of its value - you obviously can't launch 

    c:\program files (x86)\Mozilla Firefox\firefox.exe

un-quoted from a batch file. So this line will cause an error unless SLIMERJSLAUNCHER is set to a quoted string in the shell.

    SET SLIMERJSLAUNCHER="c:\program files (x86)\Mozilla Firefox\firefox.exe"

That plus the quoting at the top of slimerjs.bat ends up with the value being double-quoted, with the inner quotes considered a part of the string. Now the line launching the app works.. on the other hand, when extra quotes ARE part of the value, the IF NOT EXIST checks will fail - because the IF NOT EXIST command won't un-quote the string. So this:

 ```
if not exist %SLIMERJSLAUNCHER% (
    call :findFirefox
)
```
actually turns into 
```
if not exist ""c:\program files (x86)\Mozilla Firefox\firefox.exe""(
```
and findFirefox is always called, ignoring and overwriting the SLIMERJSLAUNCHER you set manually on the command line.

This fix https://github.com/laurentj/slimerjs/commit/c7490f275841a12400a0e0e192720637b21de527 was a step in the right direction, but incomplete. The command launching the application should also be quoted.

My suggestion: avoid getting into confusing situations where quotes are needed as part of the string value. I have an upcoming PR to fix it..

**Comment by laurentj** [comment]
Did you test slimerjs.bat on all version of Windows? Which version to you use?

Each version of Windows has a different behavior with quotes or different support of quotes.

**Comment by hallvors** [comment]
It's tested on Win8 64bit. I could do some testing on WinXP too, but I don't have Win7 or Vista handy. The behaviour is indeed very confusing to figure out, but are you sure all versions really are different?

**Comment by hallvors** [comment]
On which version of Windows has the current code been tested? Did somebody set SLIMERJSLAUNCHER and verify it was working?

Here's a test case - you can set SLJS_TEST to a valid path with or without quotes and run this .bat file to see how Windows handles it:

https://gist.github.com/hallvors/eb4a622686bf87cbb4bb

```
Path is ""c:\Program Files (x86)\Mozilla Firefox\firefox.exe""

This path DOES exist: ""c:\Program Files (x86)\Mozilla Firefox\firefox.exe""
'Files' is not recognized as an internal or external command,
operable program or batch file.
This path does NOT exist: ""c:\Program Files (x86)\Mozilla Firefox\firefox.exe""
```

Yes Windows, you're insane. It shouldn't be possible to conclude that a path *both* exists and doesn't - right? :-/

**Comment by laurentj** [comment]
>are you sure all versions really are different?

Yes. This is a nightmare. See commits history on slimerjs.bat. Developing on the command line on windows is a nightmare. I would like to drop this slimerjs.bat and force users to Install cygwin as it is easier with.

**Comment by hallvors** [comment]
Well, I have studied the commit history regarding quotes, and I've spent enough time on this to understand that it's a nasty mess - but it would suck to require cygwin because you'd exclude lots of competent but not deeply technical users..

Are you sure the main problem is the Windows *version* and not whether the installation path had spaces or not across the various locales? Quoting only becomes an issue if the path has spaces, so earlier versions of slimerjs.bat might work just fine on an XP machine with c:\programfiler\MozillaFirefox\firefox.exe but fail on the same Windows version if the path is c:\Program Files\Mozilla Firefox\firefox.exe ? 

I will test on the machines I've got and ask a few others who might have Windows machines to test too..

**Comment by abozhilov** [comment]
For better maintainability I think the .bat should be rewritten with power shell capabilities.

**Comment by hallvors** [comment]
So far, we've confirmed that the "QUOTED IN EXISTS STATEMENT" branch of the logic in that gist is sane on both WinXP and Win8 (and this happens to be the approach I'm taking in my pull request).

**Comment by hallvors** [comment]
@laurentj I'm pretty sure this PR is the right way to fix the .bat file quoting problem.. I don't have a Win7 machine handy but it's tested on Win XP and Win 8 on paths with and without quotes. Please consider taking it in.

**Comment by laurentj** [comment]
merged and backported on v0.9
